### PR TITLE
7seg driver

### PR DESCRIPTION
**Pull request by A3sanjay**

### CODE REVIEW

File: libraries/ms-drivers/seg_display_driver/src/seg_display.c [hunk 1]
```diff
@@ -0,0 +1,97 @@
+#include "seg_display.h"
+
+#include "bcd.h"
+#include "log.h"
+
```
Comment (JarvisWeng) [inline libraries/ms-drivers/seg_display_driver/src/seg_display.c:5]: Could you create a
```
static PincBcd s_bcd_pins = { ..... };
static DispPins s_disp_sel_pins = { ......};
```
Then delete `PinsBcd *bcd_pins, DispPins *disp_sel_pins` from all your function parameters and use the global variables instead.

File: libraries/ms-drivers/seg_display_driver/src/seg_display.c [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  gpio_init_pin(&s_bcd_pins.B, GPIO_ANALOG, GPIO_STATE_LOW);
+  gpio_init_pin(&s_bcd_pins.C, GPIO_ANALOG, GPIO_STATE_LOW);
+  gpio_init_pin(&s_bcd_pins.D, GPIO_ANALOG, GPIO_STATE_LOW);
+  gpio_init_pin(&s_bcd_pins.DP, GPIO_ANALOG, GPIO_STATE_LOW);
+  gpio_init_pin(&s_disp_sel_pins.D1, GPIO_ANALOG, GPIO_STATE_LOW);
+  gpio_init_pin(&s_disp_sel_pins.D2, GPIO_ANALOG, GPIO_STATE_LOW);
+  gpio_init_pin(&s_disp_sel_pins.D3, GPIO_ANALOG, GPIO_STATE_LOW);
+}
+
+void set_seg_display(uint8_t disp_value, uint8_t pin_number) {
```
Comment (mitchellostler) [inline libraries/ms-drivers/seg_display_driver/src/seg_display.c:96]: should take a SegDisplay* display as an argument

File: libraries/ms-drivers/seg_display_driver/src/seg_display.c [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  // Sets A-D with the binary digits of 8
+  uint8_t A = 1 ? (dec_to_bcd(disp_value) | 1) == dec_to_bcd(disp_value) : 0;
```
[the remark this answers is not in the document]
Comment (mitchellostler) [reply libraries/ms-drivers/seg_display_driver/src/seg_display.c:98]: You can make this much cleaner with the following 
```
// Get GPIO values
uint8_t bcd = dec_to_bcd(disp_value);
GpioState state_a = bcd & 0x1;
GpioState state_b = bcd & 0x2;
GpioState state_c = bcd & 0x4;
GpioState state_d = bcd & 0x4;
```
And then just set each of the GPIOs to the value 
````
gpio_set_state(display->A, state_a);
...
```
etc

**Review by mitchellostler — CHANGES_REQUESTED** [review]
There's quite a few comments, so I'll leave you with these ones to address, and give a second review once you get these done

File: libraries/ms-drivers/seg_display_driver/inc/seg_display.h [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+} SegDisplay;
+
+// Initializes input and display GPIOs
+StatusCode seg_display_init(SegDisplay *display);
+
+// Sets an integer value onto the display with a max value of 999
+StatusCode seg_display_set_int(SegDisplay *display, uint16_t val);
+
+// Sets a single digit decimal value onto the display with a max value of 99.9
+StatusCode set_display_set_float(SegDisplay *display, float val);
```
Comment (mitchellostler) [inline libraries/ms-drivers/seg_display_driver/inc/seg_display.h:26]: This should be seg_display instead of set_display at the start

**Review by mitchellostler — COMMENTED** [review]
Just the one comment, otherwise looks really good.

**Review by mitchellostler — APPROVED** [review]

**Pull request merged**